pkg/istio/authorizationpolicy/controller.go

```diff
@@ -0,0 +1,463 @@
+							c.ConfigStoreCache.Delete(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), obj.Name, obj.Namespace)
+						}
+					} else {
+						err := c.findDeleteDryrunResource(obj.Name, obj.Namespace)
+						if err != nil {
+							log.Errorln("error deleting local authz policy file: ", err.Error())
+							return err
+						}
+					}
+				}
+			} else {
+				// case when service has authzEnabledAnnotation removed, and authz policy with the same name present
+				if !c.Dryrun {
+					if res := c.ConfigStoreCache.Get(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), obj.Name, obj.Namespace); res != nil {
+						log.Infoln("Deleting Authz Policy ... ")
+						c.ConfigStoreCache.Delete(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), obj.Name, obj.Namespace)
+					}
+				} else {
+					err := c.findDeleteDryrunResource(obj.Name, obj.Namespace)
+					if err != nil {
+						log.Errorln("error deleting local authz policy file: ", err.Error())
+						return err
+					}
+				}
+			}
+		} else if castItem.Operation == model.EventDelete {
+			log.Infoln("Deleting Authz Policy ... ")
+			if !c.Dryrun {
+				c.ConfigStoreCache.Delete(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), obj.Name, obj.Namespace)
+			} else {
+				err := c.findDeleteDryrunResource(obj.Name, obj.Namespace)
+				if err != nil {
+					log.Errorln("error deleting local authz policy file: ", err.Error())
+					return err
+				}
+			}
+		}
+	case adCast:
+		obj, _ := (castItem.Resource).(*adv1.AthenzDomain)
+		// athenz domain create/update event should trigger a sync with existing authz policies in the corresponding namespace
+		if castItem.Operation == model.EventUpdate || castItem.Operation == model.EventAdd {
+			res, err := c.ConfigStoreCache.List(collections.IstioSecurityV1Beta1Authorizationpolicies.Resource().GroupVersionKind(), m.DomainToNamespace(obj.Name))
+			if err != nil {
+				return fmt.Errorf("Unable to list authz policies in namespace: %s", obj.Namespace)
+			}
+			for _, authzPolicy := range res {
+				signedDomain := obj.Spec.SignedDomain
+				domainRBAC := m.ConvertAthenzPoliciesIntoRbacModel(signedDomain.Domain, &c.AdIndexInformer)
+				authzSpec, ok := (authzPolicy.Spec).(*v1beta1.AuthorizationPolicy)
+				if !ok {
+					return fmt.Errorf("unable to cast interface to authorizationpolicies object")
+				}
+				convertedCR := c.RbacProviderV2.ConvertAthenzModelIntoIstioAuthzPolicy(domainRBAC, authzPolicy.Namespace, authzPolicy.Name, authzSpec.Selector.MatchLabels["svc"])
+				log.Infof("Athenz Domain %s updated, updating Authz Policy in namespace %s ... ", obj.Name, m.DomainToNamespace(obj.Name))
+				// assign current revision, update function requires a defined resource version
+				convertedCR.ResourceVersion = authzPolicy.ResourceVersion
+				if (!c.Dryrun) {
+					revision, err := c.ConfigStoreCache.Update(convertedCR)
```

Comment:
We are updating every single resource here even if it didn't have a change. In the existing controller, we calculate only the items that changed so that we can apply them: https://github.com/yahoo/k8s-athenz-istio-auth/blob/master/pkg/controller/controller.go#L180
Can we do the same here? This will avoid a lot of extra api calls.

Reply:
Sure, @CatherineZhuang, after matching with wildcard services, we can update those policies that have config change